compiler/rustc_codegen_gcc/src/consts.rs

```diff
@@ -81,7 +81,12 @@ impl<'gcc, 'tcx> StaticCodegenMethods for CodegenCx<'gcc, 'tcx> {
         if global.to_rvalue().get_type() != val_llty {
             global.to_rvalue().set_type(val_llty);
         }
-        set_global_alignment(self, global, alloc.align);
+
+        let align = match attrs.alignment {
+            Some(alignment) => Ord::max(alignment, alloc.align),
```

Comment:
Shouldn't `alloc.align` already be the right value? Seems quite error-prone if every user is expected to compensate for this separately.

Reply:
Yeah, `create_static_alloc` will need the ability to overalign the allocation, e.g. with a new separate `align` parameter.